### DISCUSSION TITLE
Codespaces / .devcontainer setup

### DISCUSSION THREAD

**Discussion by sambarfoot** [supabase/supabase#15365] [Questions]
Hi

I'm new to supabase. I've been looking around the documentation and struggling to find any examples of how to setup a supabase local environment to run either within codespaces or using vscode .devcontainer. Any help would be much appreciated?

thanks

**Comment by sambarfoot** [comment]
I found the answer using docker-in-docker

**Reply by ozanvos** [reply]
Hi @sambarfoot, were you able to get this working? Starting my Supabase dev server works but I'm getting CORS issues connecting to the auth api.

**Reply by sambarfoot** [reply]
Hi @osener are you still having issues? Sorry for the late reply, this is how I did it. It works with vsocde. With codespaces, you are missing the web front end. 

devcontainer.json
```
{
  "name": "Supabase Backend",
  // "image": "mcr.microsoft.com/devcontainers/base:bullseye",
  "build": {
    "dockerfile": "Dockerfile"
  },
  "customizations": {
    "vscode": {
      "settings": {
        "editor.tabSize": 2
      },
      "extensions": [
        "vscode.typescript-language-features",
        "VisualStudioExptTeam.vscodeintellicode",
        "GitHub.copilot",
        "Gruntfuggly.todo-tree",
        "denoland.vscode-deno"
      ]
    }
  },
  "postCreateCommand": "curl -sL https://app.snaplet.dev/get-cli/ | bash",
  "features": {
    "ghcr.io/devcontainers/features/docker-in-docker:latest": {}, // Install Docker CLI
    // "ghcr.io/devcontainers-contrib/features/deno:latest": {}, // Install Deno
    "ghcr.io/devcontainers-contrib/features/supabase-cli:latest": {} // Install Supabasez CLI
  }
} 
```

Dockerfile
```
FROM mcr.microsoft.com/devcontainers/base:bullseye

# Install deno
ENV DENO_INSTALL=/deno
RUN mkdir -p /deno \
    && curl -fsSL https://gist.githubusercontent.com/LukeChannings/09d53f5c364391042186518c8598b85e/raw/ac8cd8c675b985edd4b3e16df63ffef14d1f0e24/deno_install.sh | sh \
    && chown -R vscode /deno

ENV PATH=${DENO_INSTALL}/bin:${PATH} \
    DENO_DIR=${DENO_INSTALL}/.cache/deno
```

**Comment by tothambrus11** [comment]
I made a Supabase starter repo with devcontainer and GitHub CI working using docker-outside-of-docker: https://github.com/tothambrus11/supabase-experiment

And here is one that uses docker-in-docker: https://github.com/tothambrus11/supabase-dind
